test/validation/api/ipsec/ipsec.c

```diff
@@ -447,6 +449,12 @@ odp_packet_t ipsec_packet(const ipsec_test_packet *itp)
 
 	odp_packet_user_ptr_set(pkt, PACKET_USER_PTR);
 
+	eth_hdr = (odph_ethhdr_t *)odp_packet_l2_ptr(pkt, NULL);
+	type = odp_be_to_cpu_16(eth_hdr->type);
+	if (type == ODPH_ETHTYPE_IPV4)
+		odp_packet_has_ipv4_set(pkt, 1);
```

Comment:
This is not right. IPsec operations do not require the has_ipv4 or has_ipv6 flags to be set (see the api spec, e.g. ipsec.h:1287). The implementation is expected to look into the packet data to see whether the packet is IPv4 or IPv4. Setting the has_ipv4 or has_ipv6 flags here would make the validation test accept incorrect implementations that require the flags to be set according to the real packet type.

Reply:
Abandoning this change as pointed out, as per the spec only l3 and l4 offsets need to have valid values.